libraries/AP_BattMonitor/AP_BattMonitor_SMBus_SUI.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor_SMBus.h"
+#include <AP_HAL/I2CDevice.h>
+
+// Base SUI class
+class AP_BattMonitor_SMBus_SUI : public AP_BattMonitor_SMBus
+{
+public:
+
+    // Constructor
+    AP_BattMonitor_SMBus_SUI(AP_BattMonitor &mon,
+                    AP_BattMonitor::BattMonitor_State &mon_state,
+                    AP_BattMonitor_Params &params,
+                    AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev,
+                    uint8_t cell_count
+                    );
+
+    void init(void) override;
+
+protected:
+
+    std::vector<float> currents;
```

Comment:
Are we actually okay with letting `std::vector` in here?

Reply:
removed